src/main/kotlin/app/UserCtrl.kt

```diff
@@ -46,9 +46,13 @@ object UserCtrl {
         if (Cache.contains(username)) {
             return true
         }
-        if (GhService.remainingRequests == 0) {
+        val remainingRequests = GhService.remainingRequests
+        if (remainingRequests <= 0) {
             return false
         }
+        if (Config.starRequestBypassLevel?.let { remainingRequests - it > 0 } == true) {
```

Comment:
The function is named hasStarredRepo, so this check should probably be performed before calling it:

```unrestricted || freeRequestsLeft  || UserCtrl.hasStarredRepo(user)```